app_flutter/lib/state/brooks.dart

```diff
@@ -0,0 +1,78 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+typedef BrookCallback<T> = void Function(T event);
+
+/// A source of events of type T.
+///
+/// This is a light-weight stream, hence the name of the class.
+///
+/// Listeners cannot be registered multiple times simultaneously.
+class Brook<T> {
+  Brook();
+
+  final Set<BrookCallback<T>> _listeners = <BrookCallback<T>>{};
```

Comment:
Just a note: I don't think there's any current use cases in app_flutter where there are multiple listeners

Reply:
I expect we'll have a second listener if we add logging to the server.